x/deployment/types/types_test.go

```diff
@@ -0,0 +1,71 @@
+package types_test
+
+import (
+	"testing"
+
+	"github.com/ovrclk/akash/testutil"
+	"github.com/ovrclk/akash/x/deployment/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type gStateTest struct {
+	state                types.GroupState
+	expStateStr          string
+	expValidateOrderable error
+	expValidateClosable  error
+}
+
+func TestGroupState(t *testing.T) {
+	tests := []gStateTest{
+		{
+			state:       types.GroupOpen,
+			expStateStr: "open",
+		},
+		{
+			state:                types.GroupOrdered,
+			expStateStr:          "ordered",
+			expValidateOrderable: types.ErrGroupNotOpen,
+		},
+		{
+			state:                types.GroupMatched,
+			expStateStr:          "matched",
+			expValidateOrderable: types.ErrGroupNotOpen,
+		},
+		{
+			state:                types.GroupInsufficientFunds,
+			expStateStr:          "insufficient",
+			expValidateOrderable: types.ErrGroupNotOpen,
+		},
+		{
+			state:                types.GroupClosed,
+			expStateStr:          "closed",
+			expValidateClosable:  types.ErrGroupClosed,
+			expValidateOrderable: types.ErrGroupNotOpen,
+		},
+		{
+			state:                types.GroupState(99),
+			expStateStr:          "UNKNOWN",
+			expValidateOrderable: types.ErrGroupNotOpen,
+		},
+	}
+
+	for i, test := range tests {
+		t.Logf("------test-%d: %#v", i, test)
+		assert.Equal(t, types.GroupStateToReadable(test.state), test.expStateStr)
+		id := testutil.GroupID(t)
+		group := types.Group{
+			GroupID: types.GroupID{
+				Owner: id.Owner,
+				DSeq:  id.DSeq,
+				GSeq:  id.GSeq,
+			},
+			State: test.state,
+		}
+
+		assert.NotEmpty(t, group.ID().Owner)
+
+		assert.Equal(t, group.ValidateOrderable(), test.expValidateOrderable)
+
+		assert.Equal(t, group.ValidateClosable(), test.expValidateClosable)
```

Comment:
Any reason why you don't just use `GroupID: id`?

Look at [`assert.Equal`](https://godoc.org/github.com/stretchr/testify/assert#Equal) and [`assert.Equalf`](https://godoc.org/github.com/stretchr/testify/assert#Equalf).

1. the "expected" value is the second argument.
1. you can add messages to both `Equal` and `Equalf`

for example:

```
assert.Equal(t, test.expValidateOrderable, group.ValidateOrderable(), test.expStateStr)
```